pkgs/shells/fish/wrapper.nix

```diff
@@ -14,12 +14,12 @@ let
   complPath = completionDirs ++ map (vendorDir "completions") pluginPkgs;
   funcPath = functionDirs ++ map (vendorDir "functions") pluginPkgs;
   confPath = confDirs ++ map (vendorDir "conf") pluginPkgs;
-  safeConfPath = map escapeShellArg confPath;
 
 in writeShellScriptBin "fish" ''
   ${fish}/bin/fish --init-command "
     set --prepend fish_complete_path ${escapeShellArgs complPath}
     set --prepend fish_function_path ${escapeShellArgs funcPath}
-    for c in {${concatStringsSep "," safeConfPath}}/*; source $c; end
+    set --local fish_conf_source_path ${escapeShellArgs confPath}
+    for c in $fish_conf_source_path/*; source $c; end
```

Comment:
@pacien I may be missing something obvious, but isn't this going to break in the case where confPath is not a singleton?

Reply:
Please ignore my previous comment and thanks for helping me better understand fish cartesian product.

Reply:
I had the same question and had to run it myself to totally understand:

```
$ mkdir a b c d e
$ touch a/1 b/2 c/3 d/4 e/5
$ set --local test a b c d e
$ for c in $test/*; echo $c; end
a/1
b/2
c/3
d/4
e/5
```